common/persistence/sql/sqlConfigStore_test.go

```diff
@@ -0,0 +1,177 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package sql
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/persistence/sql/sqlplugin"
+)
+
+func TestFetchConfig(t *testing.T) {
+
+	testCases := []struct {
+		name       string
+		configType persistence.ConfigType
+		mockSetup  func(*sqlplugin.MockDB)
+		want       *persistence.InternalConfigStoreEntry
+		wantErr    bool
+	}{
+		{
+			name:       "Success case",
+			configType: persistence.DynamicConfig,
+			mockSetup: func(mockDB *sqlplugin.MockDB) {
+				mockDB.EXPECT().SelectLatestConfig(gomock.Any(), gomock.Any()).Return(&persistence.InternalConfigStoreEntry{}, nil)
+				mockDB.EXPECT().IsNotFoundError(nil).Return(false)
+			},
+			want:    &persistence.InternalConfigStoreEntry{},
+			wantErr: false,
+		},
+		{
+			name:       "Not found error",
+			configType: persistence.DynamicConfig,
+			mockSetup: func(mockDB *sqlplugin.MockDB) {
+				notFoundErr := errors.New("not found")
+				mockDB.EXPECT().SelectLatestConfig(gomock.Any(), gomock.Any()).Return(nil, notFoundErr)
+				mockDB.EXPECT().IsNotFoundError(notFoundErr).Return(true)
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name:       "Database error",
+			configType: persistence.DynamicConfig,
+			mockSetup: func(mockDB *sqlplugin.MockDB) {
+				err := errors.New("db error")
+				mockDB.EXPECT().SelectLatestConfig(gomock.Any(), gomock.Any()).Return(nil, err)
+				mockDB.EXPECT().IsNotFoundError(err).Return(false)
+				mockDB.EXPECT().IsNotFoundError(err).Return(false)
+				mockDB.EXPECT().IsTimeoutError(err).Return(true)
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockDB := sqlplugin.NewMockDB(ctrl)
+			store, err := NewSQLConfigStore(mockDB, nil, nil)
+			require.NoError(t, err)
```

Comment:
can you provide the optional 3rd parameter to these require/assert calls so test output contains more context?
I personally don't prefer these helpers and just do validation natively `if condition {t.Errorf("Got %v, want: %v")}`

Reply:
+1 for the context. I think it is well-expected to have a comment (a reason) for require/assert as the latest argument though.